jupyter_archive/handlers.py

```diff
@@ -154,24 +158,28 @@ async def get(self, archive_path, include_body=False):
         self.flush_cb = ioloop.PeriodicCallback(self.flush, self.archive_download_flush_delay)
         self.flush_cb.start()
 
-        args = (
-            archive_path,
-            archive_format,
-            archive_token,
-            follow_symlinks,
-            download_hidden,
-        )
-        await ioloop.IOLoop.current().run_in_executor(None, self.archive_and_download, *args)
-
-        if self.canceled:
-            self.log.info("Download canceled.")
-        else:
-            # Here, we need to flush forcibly to move all data from _write_buffer to stream._write_buffer
-            self.flush(force=True)
-            self.log.info("Finished downloading {}.".format(archive_filename))
+        try:
+            args = (
+                archive_path,
+                archive_format,
+                archive_token,
+                follow_symlinks,
+                download_hidden,
+            )
+            await ioloop.IOLoop.current().run_in_executor(None, self.archive_and_download, *args)
+
+            if self.canceled:
+                self.log.info("Download canceled.")
+            else:
+                # Here, we need to flush forcibly to move all data from _write_buffer to stream._write_buffer
+                self.flush(force=True)
+                self.log.info("Finished downloading {}.".format(archive_filename))
+        except Exception:
+            raise
+        finally:
+            self.flush_cb.stop()
```

Comment:
In the investigation this problem, I noticed that callback thread is not stopped when an exception occurs during archive processing. This is a fix to ensure that callback thread is stopped.
This is not directly related to this issue, so please let me know if you prefer not to include it in this PR.